### PR TITLE
refactor: Add extension column

### PR DESCRIPTION
**Pull request by PedroChaparro**
Includes: 

- Create the `extension` column to separate the file extension and name. 
- Create  the `files_view` view to ease some queries. 
- Update endpoints and tests.

### CODE REVIEW

File: src/main/resources/migrations/V1.2.0__files_view.sql [hunk 1]
```diff
@@ -0,0 +1,21 @@
+-- Add a new column to separate the extension from the name
+ALTER TABLE archives ADD COLUMN extension VARCHAR(16) NOT NULL DEFAULT '';
```
Comment (Woynert) [inline src/main/resources/migrations/V1.2.0__files_view.sql:2]: I believe you must delete the default value after adding the new column to the existing rows.

**Review by Woynert — CHANGES_REQUESTED** [review]

File: src/main/resources/migrations/V1.2.0__files_view.sql (hunk 1 above)
Comment (PedroChaparro) [reply src/main/resources/migrations/V1.2.0__files_view.sql:2]: Thanks

**Comment by PedroChaparro** [comment]
Right now extensions can only be empty when the `fileType` is equal to `directory`. I'll change that.

**Comment by PedroChaparro** [comment]
> Can't seem to be able to create a file with `null` extension. It is not guaranteed that a mimetype can be recognized from a file.
> 

Right now extensions can only be empty when the `fileType` is equal to `directory`. I'll change that.

**Review by Woynert — APPROVED** [review]
Well done!

**Pull request merged**